addons/im_livechat/static/src/embed/common/livechat_button.js

```diff
@@ -43,6 +43,17 @@ export class LivechatButton extends Component {
             },
         });
         useExternalListener(document.body, "scroll", this._onScroll, { capture: true });
+        useEffect(
+            () => {
+                const rootNodeClassList = this.ref.el?.getRootNode().host?.classList;
+                if (rootNodeClassList) {
+                    this.isShown
+                        ? rootNodeClassList.add("livechat-button-visible")
+                        : rootNodeClassList.remove("livechat-button-visible");
+                }
+            },
+            () => [this.isShown]
+        );
```

Comment:
Nitpick: I would name the classname `o-livechat-LivechatButton-isVisible`

Reasoning:
- prefix `o` to reduce conflicts with embed livechat for the most part
- discuss BEM convention for idiomatic traceability of where/who defines this classname (-> the `LivechatButton` component)
- `isVisible` has a sort of part-name / modifier. `isVisible` gives a grasp of the semantics of such classname

Reply:
> We do our best with the possibilities at hand.

It's not like we are going to have something better at some point.

> his looks unconventional for purists wanting exclusively camelCase or kebab-case or PascalCase, but the mixture is intentionally chosen for keeping most things straight-forward.

This notation is anything but straightforward. My initial naming was more straightforward than this. This notation packs way too much information. And the usefulness of this information isn't obvious to me when you will grep the class anyway…

Reply:
- `o-` to prefix odoo classnames
- `livechat` for module name
- `LivechatButton` for component name
- `isVisible` for the part name

Each chunk is joined with `-` as expected with classname format.

This is the most straight-forward and idiomatic way to name a classname that has good traceability, no conflict with other component and so on.